packages/core/src/__tests__/dispatcher.test.ts

```diff
@@ -35,6 +35,16 @@ const ipfs = {
   id: async () => ({ id: 'ipfsid' }),
 }
 
+const recordCache = {
+  get: jest.fn(async (cid: CID, record: any) => {
+    // CID not found in cache
+    if (cid === FAKE_CID2) return null
```

Comment:
you need to use `.equals` to compare CIDs for equality.  `==` will use object equality - i.e. checking if you the pointers are to the same CID instance in memory.